dart/dynamics/BodyNode.cpp

```diff
@@ -896,6 +819,60 @@ const Joint* BodyNode::getChildJoint(size_t _index) const
 }
 
 //==============================================================================
+ShapeNode* BodyNode::createShapeNode(const ShapePtr& shape,
+                                     const std::string& name)
+{
+  ShapeNode::Properties properties;
+  properties.mShape = shape;
+
+  return createNode<ShapeNode>(properties);
+}
+
+//==============================================================================
+ShapeNode* BodyNode::createShapeNode(const ShapePtr& shape, const char* name)
+{
+  return createShapeNode(shape, std::string(name));
+}
+
+//==============================================================================
+size_t BodyNode::getNumShapeNodes() const
+{
+  return getNumNodes<ShapeNode>();
+}
+
+//==============================================================================
+ShapeNode* BodyNode::getShapeNode(size_t index)
+{
+  return getNode<ShapeNode>(index);
+}
+
+//==============================================================================
+const ShapeNode* BodyNode::getShapeNode(size_t index) const
+{
+  return getNode<ShapeNode>(index);
+}
+
+//==============================================================================
+std::vector<ShapeNode*> BodyNode::getShapeNodes()
+{
+  std::vector<ShapeNode*> shapeNodes(getNumShapeNodes());
+
+  auto numShapeNode = getNumShapeNodes();
+  for (auto i = 0u; i < numShapeNode; ++i)
+    shapeNodes[i] = getShapeNode(i);
+
+  return shapeNodes;
+}
```

Comment:
My concern was this case:
```cpp
auto shapeNodes1 = bodyNode->getShapeNodes();  // vector gets created
// change number of shape nodes of bodyNode
auto shapeNodes2 = bodyNode->getShapeNodes();  // another vector gets created
```

If we have the vector as a member variable of BodyNode, then the vector is created once when BodyNode instance is created no matter whether the num of shape nodes is changed after. Probably we would be able to avoid memory allocation if the new number of shape nodes is less than the allocated memory size (of course it depends on the implementation of vector).